events/sql.py

```diff
@@ -34,4 +35,41 @@ def count_events_for_keywords(keyword_ids=()):
             ) t
             GROUP BY t.keyword_id;
             ''')
+        else:
+            return {}
+        return dict(cursor.fetchall())
+
+
+def count_events_for_places(place_ids=(), all=False):
+    """
+    Get the actual count of events in the given places.
+
+    :param place_ids: set of place ids
+    :param all: count all places instead
+    :type place_ids: Iterable[str]
+    :return: dict of place id to count
+    :rtype: dict[str, int]
+    """
+    # sorry for the non-DRY-ness; would be easier with an SQL generator like SQLAlchemy, but...
+
+    place_ids = tuple(set(place_ids))
+    with connection.cursor() as cursor:
+        if place_ids:
+            cursor.execute('''
+            SELECT t.location_id, COUNT(DISTINCT t.id)
+            FROM (
+              SELECT location_id, id FROM events_event WHERE location_id IN %s
```

Comment:
Place event counts don't need a subquery, since they need to pick up data from a single table only.

```sql
SELECT e.location_id, COUNT(*)
FROM events_event e
-- possible where clause here
GROUP BY e.location_id;
```